repo/blob/sftp/sftp_storage_test.go

```diff
@@ -4,23 +4,130 @@ import (
 	"context"
 	"fmt"
 	"io/ioutil"
+	"net"
 	"os"
+	"os/exec"
+	"path/filepath"
 	"strconv"
+	"strings"
 	"testing"
+	"time"
 
 	"github.com/kopia/kopia/internal/blobtesting"
 	"github.com/kopia/kopia/internal/testlogging"
+	"github.com/kopia/kopia/internal/testutil"
 	"github.com/kopia/kopia/repo/blob"
 	"github.com/kopia/kopia/repo/blob/sftp"
 )
 
+const (
+	dockerImage  = "atmoz/sftp"
+	dialTimeout  = 10 * time.Second
+	sftpUsername = "foo"
+)
+
+func mustGetLocalTmpDir(t *testing.T) string {
+	t.Helper()
+
+	tmpDir, err := ioutil.TempDir(".", ".creds")
```

Comment:
Why doesn't `t.TempDir()` work here?

Reply:
for some reason the permissions/mountpoints of `t.TempDir()` are somehow messing with docker. at least did not work for me on a mac.

Reply:
got it, thanks the for explanation. It'd be good to include a comment as others or our future selves may wonder the same.